lib/io/i2c.cpp

```diff
@@ -16,7 +16,7 @@ namespace hyped::io {
 I2c::I2c(const std::uint8_t bus_address, core::ILogger &log) : sensor_address_(0), log_(log)
 {
   char path[13];  // up to "/dev/i2c-2"
-  sprintf(path, "/dev/i2c-%d", bus_address);
+  snprintf(path, sizeof(path), "/dev/i2c-%d", bus_address);
```

Comment:
This has been removed in main branch so just merge master into this branch to resolve merge conflicts